src/main/java/frc/robot/subsystems/Indications.java

```diff
@@ -1,33 +1,82 @@
 package frc.robot.subsystems;
 
+
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 
+
 //LEDS
 import edu.wpi.first.wpilibj.AddressableLED;
 import edu.wpi.first.wpilibj.AddressableLEDBuffer;
 
+
 //Constants
 import frc.robot.Constants.kSensors;
 
+
 public class Indications extends SubsystemBase {
 
-  private AddressableLED m_LedStrip;
-  private AddressableLEDBuffer m_LedData;
-
-  public Indications() {
-    m_LedStrip = new AddressableLED(kSensors.ledPort);
-    m_LedData = new AddressableLEDBuffer(kSensors.ledLength);
-    m_LedStrip.setData(m_LedData);
-    m_LedStrip.start();
-  }
-
-  @Override
-  public void periodic() {
-    // This method will be called once per scheduler run
-  }
-
-  @Override
-  public void simulationPeriodic() {
-    // This method will be called once per scheduler run during simulation
-  }
+
+ public class RGB{
+   public int red;
+   public int green;
+   public int blue;
+
+
+
+
+   public RGB(int red, int green, int blue){
+     this.red = red;
+     this.green = green;
+     this.blue = blue;
+   }
+
+
+ }
+
+
+ private AddressableLED m_LedStrip;
+ private AddressableLEDBuffer m_LedData;
+
+
+ public Indications() {
+   m_LedStrip = new AddressableLED(kSensors.ledPort);
+   m_LedData = new AddressableLEDBuffer(kSensors.ledLength);
+   m_LedStrip.setData(m_LedData);
+   m_LedStrip.start();
+ }
+
+
+ public void pattern2(RGB color1,RGB color2, float colorSpeed){
+   // 2 Colored Pattern
+    var int patternSize = 2;
+   for (var i = 1; i < m_ledData.getLength(); i++) {

```

Comment:
You also need to declare the `offset` variable that is used below in the `for()` loop and after. Do that just before the loop, like this.
```suggestion
    int offset = 0;
    for (var i = 1; i < m_ledData.getLength(); i++) {
```